hid/testHID.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+
+'''
+This will output:
+Hemiola
+works
+To the device the raspberry pi is connected to.
+'''
+
+NULL_CHAR = chr(0)
+
+def write_report(report):
+    with open('/dev/hidg0', 'rb+') as fd:
+        fd.write(report.encode())
+
```

Comment:
I think you could simplify this by doing something like:
```
for char in "Hemiola\nworks":
    write_report( NULL_CHAR*2 + char + NULL_CHAR*5 )
```
Unless this is also intended to be an example of how to press keys etc.

Reply:
It is an example of how key presses work, but also I am not printing the `char` directly. Those are the escape codes. Also, notice that the lower case key codes come first, which is not how it works in ascii.
You can checkout the inspiration [here](https://randomnerdtutorials.com/raspberry-pi-zero-usb-keyboard-hid/).

Reply:
Ah, I didn't realize they didn't directly convert to ascii.  I still think it could be compacted to a lambda/function but either way makes it nice example of how to write some bytes out to the host as a HID.

Code looks good but I'm waiting on similar hardware to test it myself. If this is a blocker, or frankly a moderate inconvenience, for future development don't wait on me.  I'll have upcoming time to allocate to actually writing some code.

Marking resolved: not able to test it myself but don't want to bottleneck.  We can always test again once I have proper hardware and merge a subsequent patch to make any fixes.

Semi-related:
If there is a direct mapping from `{key->byte}` it might make life easier to just make a 256 byte lookup table from `{keyboard_key->ascii_byte}` at some point.  Then arbitrary keyboard remappings can be done with no performance overhead.

Reply:
This python script is not very important. It is purely for people to verify they set things up correctly.